rmw_dds_common/include/rmw_dds_common/qos.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RMW_DDS_COMMON__QOS_HPP_
+#define RMW_DDS_COMMON__QOS_HPP_
+
+#include "rmw/qos_profiles.h"
+#include "rmw/types.h"
+
+#include "rmw_dds_common/visibility_control.h"
+
+namespace rmw_dds_common
+{
+
+/// Check if two QoS profiles are compatible
+/**
+ * Two QoS profiles are compatible if a publisher and subcription
```

Comment:
TBH, I can't recall why that terminology. I'm just following suite with the terminology used throughout the rest of the ROS 2 stack, though I think we're also inconsistent about what terminology we're using :upside_down_face: The tutorials seem to use "subscriber" instead of "subscription".

Reply:
@wjwwood Probably has some insight.

Reply:
A publisher publishes, a subscription is a receipt of a subscription, it isn't a subscriber. So subscriber is the wrong term. A node is kind of a subscribe as it is the thing that subscribes. A publication could be a thing, but because our object publishes, it is a publisher.

Reply:
@wjwwood  Interesting. I guess I just think of it more like this (probably attracted to the symmetry/uniformity):

<img src="https://user-images.githubusercontent.com/865395/108788834-bf655780-752d-11eb-843c-4a1b3db49175.png" width="300"/>

Anyways, thanks for the info. It helps to fill in some of the blanks.